File: 프로그래머스/0/181831. 특별한 이차원 배열 2/특별한 이차원 배열 2.py
def solution(arr):
    # method1
    # answer=1
    # for i in range(len(arr)):
    #     for j in range(len(arr[i])):
    #         answer*=1 if arr[i][j]==arr[j][i] else 0
    # method2
    # r=int(all(arr[i][j]==arr[j][i] for i in range(len(arr)) for j in range(len(arr[i]))))
    # method3
    r=int(arr==list(map(list,zip(*arr))))
    return r

File: 프로그래머스/0/181831. 특별한 이차원 배열 2/특별한 이차원 배열 2.cpp
#include <string>
#include <vector>
#include <iostream>
using namespace std;

int solution(vector<vector<int>> arr) {
    int r=1;
    for(int i=0;i<arr.size();i++) {
        for(int j=0;j<arr[i].size();j++) {
            if(arr[i][j]!=arr[j][i]) {
                r=0;
                break;
            }
        }
    }
    return r;
}